spec.yaml

```diff
@@ -604,29 +608,32 @@ definitions:
                 srcUri:
                   type: string
                   description: |
-                    An URI to access the remote resource. The URI MAY be relative,
-                    in which case the prefix exposed by the `/ocm-provider` endpoint MUST
-                    be used, or it may be absolute (recommended). Additionally, the
-                    URI MAY include a secret hash in the path.
+                    An absolute URI to access the resource at the sending server.
                 size:
                   type: integer
+                  description: |
+                    The size of the file to be transferred from the sending server.
+          patternProperties:
+            "^.*$":
+              type: object
+              description: |
+                An optional additional protocol supported for this resource, with any custom
+                payload. Appropriate capabilities MUST be advertised in order for
+                a sender to ensure the recipient can parse such customized payloads.
         example:
           singleProtocolLegacy:
             name: "webdav"
             options:
               sharedSecret: "hfiuhworzwnur98d3wjiwhr"
               permissions: "some permissions scheme"
           singleProtocolNew:
-            name: "webdav"
-            options:
-              sharedSecret: "hfiuhworzwnur98d3wjiwhr"
+            name: "multi"
```

Comment:
Is this correct? Should maybe be `single`?

Reply:
You're right that `single` would be more accurate, but putting that value here for this case would just force implementers to add extra if statements, and not convey any information that cannot already be derived by simply counting the keys in the object. If we just always set `multi` as the name by default, then it becomes easy to ignore the name field and just look at the actual entries that are present.

Reply:
So I think `multi` is correct here.